### PR TITLE
creates a modal for user sign up, modifies routes and controllers to …

### PR DESCRIPTION
**Pull request by csaument**
…allow signup via modal, fetch operation successful, modifies controller to log user in after signup, test ensures modal renders with elements and callback functions work are called when buttons are clicked

### CODE REVIEW

File: app/javascript/components/components/Header.js [hunk 1]
```diff
@@ -11,16 +11,15 @@ import {
...
-  new_user_route,
-  sign_in_route,
-  sign_out_route,
-}) => {
+const Header = ({toggleSignUp, logged_in, sign_in_route, sign_out_route}) => {
   const [isOpen, setIsOpen] = useState(false);
   const toggle = () => setIsOpen(!isOpen);
 
+  const signUpClick = () => {
+    console.log({header: "click"})
```
Comment (Genemichael25) [inline app/javascript/components/components/Header.js]: Do we need this console log?

File: app/javascript/components/App.js [hunk 2]
```diff
@@ -24,6 +25,34 @@ const App = (props) => {
...
+  const toggleUpdateUser = () => {
+    setShowUpdateUser(!showUpdateUser)
+  }
+
+  const createUser = (user) => {
+    // const token = document.querySelector('meta[name="csrf-token"]').content
+    fetch('/users/signup', {
+      body: JSON.stringify(user),
+      headers: {
+        // "X-CSRF-Token": token,
```
Comment (Genemichael25) [inline app/javascript/components/App.js]: Do you need the token?

**Review by Genemichael25 — CHANGES_REQUESTED** [review]

File: app/javascript/components/App.js (hunk 2 above)
Comment (csaument) [reply app/javascript/components/App.js]: Normally, as part of deployment, we would turn cross-sight forgery checks back on. That's currently in our icebox, and those lines were from the example/template provided from some reseach.

File: app/javascript/components/components/Header.js (hunk 1 above)
Comment (csaument) [reply app/javascript/components/components/Header.js]: Console log was used to identify issues with processing of fetch requests for user registration. Removed and committed to branch.

**Review by Genemichael25 — APPROVED** [review]

**Pull request merged**